RedHatInsights/curiosity-frontend
# fix(inventoryListSelectors): issues/421 combine query

## What's included

- fix(openshiftView,rhelView): issues/421 viewId to productLabel
   * openshiftView,rhelView, viewId to productLabel, avoid conflicts
   * c3graphCard, annotation for viewId to productLabel
   * graphCard,Tooltip,Legend, product,shortLabel to productLabel
   * pageHeader, viewId to productLabel
- fix(inventoryList): issues/421 correct prop attr
- fix(inventoryListSelectors): issues/421 combine query
  * inventoryListSelectors, combine props, productId, viewId queries
  * openshiftView, rhelView, initialInventoryQuery
  * pagination, consolidate limit and page dispatches
  * paginationHelpers, removed
  * toolbar, replace paginationHelpers with clear all offset
  * viewReducer, queryTypes, added new action type

### Notes

- Corrects the additional issue of a mislabeled prop for "inventoryList"
- Splits out the use of the "viewId" into "productLabel". This is to avoid ID overlap between the product view itself and any architectures or sub-categories. For example the viewId from the sub-category "all" is RHEL... conflicting with RHEL the global viewId
- Slightly alters the behavior of the paging limit and offset for inventory
   - Product architectures/sub-categories are fully independent in their limit and page offset. 
      - ALL architectures/sub-categories should have limit and offset reset by the following filter selections and clearing: SLA, USAGE, UOM
         - There is the additional facet of this resetting the product/viewId limit and offset, this is expected. However, we've built into the Platform behavior of each product reseting ALL state when a user moves between products. **This is a note in the event future left menu changes alter current Platform behavior.**
      - ALL architectures/sub-categories should maintain their own page limit and offset unless otherwise reset. This means if a user moves between architectures/sub-categories GUI state should restore a prior page limit selection, unless it was reset

## How to test

It would be advisable to run through prior inventory reset bugs/issues around "offset"
See
- #413 #412 
- #405 




### Proxy run check
1. update the NPM packages with `$ yarn`
1. make sure Docker is running, plus on network, then
1. `$ yarn start:proxy`
1. For RHEL product architectures
   - Select an account that has multiple inventory entries, minimum 11 entries
   - Select a product architecture, move the paging to the next page
   - Select a different product architecture
      - Confirm the paging for this architecture does not match the prior page, this is expected
      - move the paging to the next page
   - Re-select the prior product architecture, the paging, if you have NOT selected any product filters such as SLA, UOM, or USAGE should restore to its prior page, this is expected
   - Re-select the other product architecture, the paging, if you have NOT selected any product filters such as SLA, UOM, or USAGE should restore to its prior page, this is expected
    


## Example

...

## Updates issue/story

#417 #421
